src/resolvers/taxonomyResolvers.ts

```diff
@@ -0,0 +1,259 @@
+    context: ContextWithLoaders,
+  ): Promise<GQLNode[]> {
+    const nodes = await queryNodes(
+      {
+        contentURI: contentUri,
+        key: metadataFilterKey,
+        value: metadataFilterValue,
+        ids: ids,
+        includeContexts: true,
+        filterProgrammes: true,
+        language: context.language,
+      },
+      context,
+    );
+    const filtered = filterVisible ? nodes.filter((node) => node.contexts.find((context) => context.isVisible)) : nodes;
+    return filtered.map((node) => {
+      return nodeToTaxonomyEntity(node, context.language);
+    });
+  },
+  async nodeResource(
+    _: any,
+    { id, rootId, parentId }: GQLQueryNodeResourceArgs,
+    context: ContextWithLoaders,
+  ): Promise<GQLNode> {
+    const resource = await fetchNode({ id }, context);
+    const visibleCtx = resource.contexts
+      .filter((c) => c.isVisible)
+      .filter((c) => !c.rootId.startsWith("urn:programme"));
+    const subjectCtx = rootId ? visibleCtx.filter((c) => c.rootId === rootId) : visibleCtx;
+    const topicCtx = parentId ? subjectCtx.filter((c) => c.parentIds.includes(parentId)) : subjectCtx;
+
+    const path = topicCtx?.[0]?.path || resource.path;
+    const rank = topicCtx?.[0]?.rank;
+    const contextId = topicCtx?.[0]?.contextId;
+    const relevanceId = topicCtx?.[0]?.relevanceId || "urn:relevance:core";
+    const entity = nodeToTaxonomyEntity({ ...resource, contexts: visibleCtx }, context.language, contextId);
+    return { ...entity, contextId, path, rank, relevanceId };
+  },
+  async nodeCollection(
+    _: any,
+    { language }: GQLQueryNodeCollectionArgs,
+    context: ContextWithLoaders,
+  ): Promise<GQLNode[]> {
+    const nodes = await context.loaders.nodesLoader.load({ metadataFilter: { key: "language", value: language } });
+    return nodes.map((node) => nodeToTaxonomyEntity(node, context.language));
+  },
+  async articleNode(
+    _: any,
+    { articleId, nodeId }: GQLQueryArticleNodeArgs,
+    context: ContextWithLoaders,
+  ): Promise<GQLNode | null> {
+    const node = articleId
+      ? await fetchNodeByContentUri(`urn:article:${articleId}`, context)
+      : nodeId
+        ? await fetchNode({ id: nodeId }, context)
+        : null;
+    if (!node) return null;
+
+    const visibleCtx = node.contexts.filter((c) => c.isVisible);
+    const entity = nodeToTaxonomyEntity({ ...node, contexts: visibleCtx }, context.language);
+    return {
+      ...entity,
+      contextId: visibleCtx?.[0]?.contextId,
+      rank: visibleCtx?.[0]?.rank,
+      relevanceId: visibleCtx?.[0]?.relevanceId || "urn:relevance:core",
+    };
+  },
+};
+
+export const resolvers = {
+  TaxonomyContext: {
+    async crumbs(taxonomyContext: TaxonomyContext, _: any, context: ContextWithLoaders): Promise<GQLTaxonomyCrumb[]> {
+      const parentNodes = await context.loaders.nodesLoader.loadMany(
+        taxonomyContext.parentContextIds.map((contextId) => ({ contextId })),
+      );
```

Comment:
Dette er et ganske tungt kall for å hente ut brødsmulestier, føler jeg. Kan dette addresseres i tax?

Reply:
Det kan, men eg tenkte dette var enklere. Spesielt siden url i tax regnes ut på direkten. Kan sjå om eg finn på noko.

Reply:
...Fant du på noe?

Reply:
Nei, ikkje endå.

Reply:
Oppdaterer taksonomi til å regne ut dette for oss!